File: files/300-399/324/CH-324.py
import pandas as pd
import re

path = "300-399/324/CH-324 Text Cleaning.xlsx"
input_df = pd.read_excel(path, usecols="B", skiprows=1, nrows=7)
test_df = pd.read_excel(path, usecols="D", skiprows=1, nrows=7).rename(columns=lambda col: col.replace('.1', ''))

input_df["Level"] = input_df["Level"]\
    .apply(lambda v: re.sub(r",$", "", re.sub(r"(Under Ground|Upper Ground|Ground)", r"\1,", str(v))) if pd.notna(v) else v)
print(input_df["Level"].equals(test_df["Level"])) # True

File: files/300-399/324/CH-324.R
library(tidyverse)
library(readxl)

path = "300-399/324/CH-324 Text Cleaning.xlsx"
input = read_excel(path, range = "B2:B9")
test  = read_excel(path, range = "D2:D9")

input$Level = str_remove(str_replace_all(input$Level, "(Under Ground|Upper Ground|Ground)", "\\1,"), ",$")

all.equal(input$Level, test$Level)
